src/charm.py

```diff
@@ -142,12 +142,27 @@ def __init__(self, *args):
         self._observer.start_observer()
         self._grafana_agent = COSAgentProvider(
             self,
-            metrics_endpoints=[
-                {"path": "/metrics", "port": METRICS_PORT},
+            metrics_endpoints=[{"path": "/metrics", "port": METRICS_PORT}],
+            scrape_configs=self.patroni_scrape_config,
+            refresh_events=[
+                self.on[PEER].relation_changed,
+                self.on.secret_changed,
+                self.on.secret_remove,
             ],
             log_slots=[f"{POSTGRESQL_SNAP_NAME}:logs"],
         )
 
+    def patroni_scrape_config(self) -> List[Dict]:
+        """Generates scrape config for the Patroni metrics endpoint."""
+        return [
+            {
+                "metrics_path": "/metrics",
+                "static_configs": [{"targets": ["localhost:8008"]}],
```

Comment:
why not {{ self_ip }} here instead of localhost? :-)

Reply:
It should work, but last time we discussed that, I think COS preferred to only scrape localhost. I'll ping them on MM to recheck.

Reply:
Sounds like it's OK to do that. Retested locally and all seems to be in order.